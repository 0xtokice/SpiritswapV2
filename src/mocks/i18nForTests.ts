/* istanbul ignore next */
export const mockUseTranslation = () => ({
  t: key => key,
  i18n: {
    language: undefined,
  },
});
