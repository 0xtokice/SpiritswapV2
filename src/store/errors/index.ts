import { createSlice } from '@reduxjs/toolkit';
import { ErrorState } from './errors';

const initialState: ErrorState = {
  unexpectedError: false,
};

const errorReducers = createSlice({
  name: 'errorReducer',
  initialState,
  reducers: {
    setUnexpectedError: (state, action) => {
      state.unexpectedError = action.payload;
    },
  },
});

// Actions
export const { setUnexpectedError } = errorReducers.actions;
export default errorReducers.reducer;
