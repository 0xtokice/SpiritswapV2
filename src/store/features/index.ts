import { createSlice } from '@reduxjs/toolkit';
import { FeatureFlags } from './features';

const initialState: FeatureFlags = {
  environment: `${process.env.NODE_ENV}`,
  development: {
    apemode: false,
  },
  production: {
    apemode: false,
  },
};

const featureReducers = createSlice({
  name: 'featureReducer',
  initialState,
  reducers: {},
});

export default featureReducers.reducer;
