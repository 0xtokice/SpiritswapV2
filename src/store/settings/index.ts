import { createSlice } from '@reduxjs/toolkit';
import { Settings } from './settings';
import { isMobile } from 'utils/isMobile';

const initialState: Settings = {
  notifications: !isMobile(),
  suggestions: true,
  keepSuggestions: false,
};

const settingsReducer = createSlice({
  name: 'settingsReducer',
  initialState,
  reducers: {
    setUserNotifications: (state, { payload }) => {
      state.notifications = payload.notifications;
    },
    setUserSuggestions: (state, { payload }) => {
      state.suggestions = payload.suggestions;
    },
    setKeepSuggestions: (state, { payload }) => {
      state.keepSuggestions = payload.keepSuggestions;
    },
  },
});

export const { setUserNotifications, setUserSuggestions, setKeepSuggestions } =
  settingsReducer.actions;

export default settingsReducer.reducer;
