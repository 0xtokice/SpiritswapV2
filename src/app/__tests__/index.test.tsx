import { createRenderer } from 'react-test-renderer/shallow';
import { App } from '../index';
import { mockUseTranslation } from 'mocks/i18nForTests';
import { Provider } from 'react-redux';
import { store } from 'store';

jest.mock('react-i18next', () => ({
  useTranslation: () => mockUseTranslation(),
}));

const renderer = createRenderer();

describe('<App />', () => {
  xit('should render and match the snapshot', () => {
    renderer.render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
