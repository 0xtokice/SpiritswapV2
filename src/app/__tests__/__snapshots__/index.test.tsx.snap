// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(Symbol.observable): [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <App />
</Context.Provider>
`;
